### PR TITLE
Migrate to null-safety

### LINKED ISSUE #57: Support null safety
This is a request to add support for [null safety](https://dart.dev/null-safety) to `package:flutter_sticky_header`.
We depend on your awesome package, so would be great to have null safety enabled.

The Dart/Flutter team already encourages publishing the migrated packages: See [this blog post](https://medium.com/dartlang/preparing-the-dart-and-flutter-ecosystem-for-null-safety-e550ce72c010).

See the [migration guide](https://dart.dev/null-safety/migration-guide) for details about enabling null safety.

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
This PR uses @John-Dormevil's branch https://github.com/John-Dormevil/value_layout_builder/tree/null-safety (see PR https://github.com/letsar/value_layout_builder/pull/2) for now.

I will change this to a stable version as soon as it is merged.

Fixes #57

### CODE REVIEW

**Comment by ThexXTURBOXx** [comment]
@letsar Thanks for merging https://github.com/letsar/value_layout_builder/pull/2!

My branch is already updated to use the new version of `value_layout_builder`. So, a review would be highly appreciated :)

File: lib/src/rendering/sliver_sticky_header.dart [hunk 1]
```diff
@@ -86,55 +87,53 @@ class RenderSliverStickyHeader extends RenderSliver with RenderSliverHelpers {
...
     }
     if (child != null) {
-      result.add(child.toDiagnosticsNode(name: 'child'));
+      result.add(child!.toDiagnosticsNode(name: 'child'));
     }
     return result;
   }
 
-  double computeHeaderExtent() {
+  double? computeHeaderExtent() {
```
Comment (letsar) [inline lib/src/rendering/sliver_sticky_header.dart:125]: If I read this correctly, the value cannot be null here, so we could make it a `double `and not a `double?`

File: lib/src/widgets/sliver_sticky_header.dart [hunk 2]
```diff
@@ -168,17 +165,17 @@ class SliverStickyHeader extends RenderObjectWidget {
   /// If a [StickyHeaderController] is not provided, then the value of
   /// [DefaultStickyHeaderController.of] will be used.
   SliverStickyHeader.builder({
-    Key key,
-    SliverStickyHeaderWidgetBuilder builder,
-    Widget sliver,
+    Key? key,
+    SliverStickyHeaderWidgetBuilder? builder,
```
Comment (letsar) [inline lib/src/widgets/sliver_sticky_header.dart:169]: builder should not be null

File: pubspec.yaml [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
 
 environment:
-  sdk: ">=2.2.0 <3.0.0"
-  flutter: ">=1.20.0 <2.0.0"
+  sdk: '>=2.12.0 <3.0.0'
+  flutter: '>=2.0.0 <3.0.0'
```
Comment (letsar) [inline pubspec.yaml]: pub does not want an upper bound for flutter

**Review by letsar — CHANGES_REQUESTED** [review]
Thank you a lot for this PR 🙂 ! Everthing seems fine except small minor changes.

**Comment by ThexXTURBOXx** [comment]
Very good eyes, thanks! :) I resolved the current discussions. Is there anything else you may have noticed? ^^

**Review by letsar — APPROVED** [review]
LGTM. Thanks a lot!!

**Pull request merged**